[Buttons] New Dynamic Type curves

ianegordon opened the pull request:
Add support for the new dynamic type curves.

Note that the new curve values are identical to the old curve values, just using the new API.

Large / Default:
![MDCButton-L](https://user-images.githubusercontent.com/1121006/55108582-468fc380-50aa-11e9-9ae5-c1440ab64e8e.png)

XXS:
![MDCButton-XXS](https://user-images.githubusercontent.com/1121006/55108534-30820300-50aa-11e9-9193-7943e1aac35f.png)

XXXL:
![MDCButton-XXXL](https://user-images.githubusercontent.com/1121006/55108543-34158a00-50aa-11e9-9765-ed84139f8431.png)

Commit:
[Button] New Dynamic Type curves

yarneo commented on components/Buttons/src/MDCButton.m:
```diff
@@ -819,7 +819,11 @@ - (void)updateTitleFont {
     // Dynamic type is enabled so apply scaling
-    font = [font mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleButton scaledForDynamicType:YES];
+    if (font.mdc_scalingCurve && !_mdc_legacyFontScaling) {
+      font = [font mdc_scaledFontForCurrentSizeCategory];
```
is there no need to provide what style to give the font?

ianegordon replied:
We do a little earlier in the Sample.
This font is set by the themer, which takes it from containerScheme.typographyScheme.button.

In the case of the new scheme, that font will have an mdc_scalingCurve attached.

yarneo replied:
ah awesome

material-automation commented on components/Buttons/src/MDCButton.m:
```diff
@@ -819,7 +819,11 @@ - (void)updateTitleFont {
+    if (font.mdc_scalingCurve && !_mdc_legacyFontScaling) {
+      font = [font mdc_scaledFontForCurrentSizeCategory];
+    } else {
+      font = [font mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleButton scaledForDynamicType:YES];
```
clang-format suggested the following change:


```
-      font = [font mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleButton scaledForDynamicType:YES];
+      font = [font mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleButton
+                                scaledForDynamicType:YES];
```

<!-- identifier: clang-format -->
<!-- File: components/Buttons/src/MDCButton.m -->

material-automation commented:
clang-format recommended changes.

Consider installing [git-clang-format](https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format) and running the following command:

```bash
git clang-format $(git merge-base origin/develop HEAD)
```

<!-- identifier: clang-format -->

material-automation commented:
bazel detected changes to the following targets:

```
//components/AppBar:snapshot_tests
//components/AppBar:snapshot_tests_test_bundle
//components/AppBar:unit_tests
//components/AppBar:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/AppBar:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/AppBar:unit_tests_IPHONE_5_IN_8_1
//components/AppBar:unit_tests_IPHONE_5_IN_8_1_test_bundle
//components/AppBar:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/AppBar:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/AppBar:unit_tests_IPHONE_X_IN_11_0
//components/AppBar:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/BottomAppBar:snapshot_tests
//components/BottomAppBar:snapshot_tests_test_bundle
//components/BottomAppBar:unit_tests
//components/BottomAppBar:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/BottomAppBar:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/BottomAppBar:unit_tests_IPHONE_5_IN_8_1
//components/BottomAppBar:unit_tests_IPHONE_5_IN_8_1_test_bundle
//components/BottomAppBar:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/BottomAppBar:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/BottomAppBar:unit_tests_IPHONE_X_IN_11_0
//components/BottomAppBar:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/ButtonBar:snapshot_tests
//components/ButtonBar:snapshot_tests_test_bundle
//components/ButtonBar:unit_tests
//components/ButtonBar:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/ButtonBar:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/ButtonBar:unit_tests_IPHONE_5_IN_8_1
//components/ButtonBar:unit_tests_IPHONE_5_IN_8_1_test_bundle
//components/ButtonBar:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/ButtonBar:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/ButtonBar:unit_tests_IPHONE_X_IN_11_0
//components/ButtonBar:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/Buttons:snapshot_tests
//components/Buttons:snapshot_tests_test_bundle
//components/Buttons:unit_tests
//components/Buttons:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Buttons:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Buttons:unit_tests_IPHONE_5_IN_8_1
//components/Buttons:unit_tests_IPHONE_5_IN_8_1_test_bundle
//components/Buttons:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Buttons:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Buttons:unit_tests_IPHONE_X_IN_11_0
//components/Buttons:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/Dialogs:snapshot_tests
//components/Dialogs:snapshot_tests_test_bundle
//components/Dialogs:unit_tests
//components/Dialogs:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Dialogs:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Dialogs:unit_tests_IPHONE_5_IN_8_1
//components/Dialogs:unit_tests_IPHONE_5_IN_8_1_test_bundle
//components/Dialogs:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Dialogs:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Dialogs:unit_tests_IPHONE_X_IN_11_0
//components/Dialogs:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/NavigationBar:snapshot_tests
//components/NavigationBar:snapshot_tests_test_bundle
//components/NavigationBar:unit_tests
//components/NavigationBar:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/NavigationBar:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/NavigationBar:unit_tests_IPHONE_5_IN_8_1
//components/NavigationBar:unit_tests_IPHONE_5_IN_8_1_test_bundle
//components/NavigationBar:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/NavigationBar:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/NavigationBar:unit_tests_IPHONE_X_IN_11_0
//components/NavigationBar:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/Snackbar:snapshot_tests
//components/Snackbar:snapshot_tests_test_bundle
//components/Snackbar:unit_tests
//components/Snackbar:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Snackbar:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Snackbar:unit_tests_IPHONE_5_IN_8_1
//components/Snackbar:unit_tests_IPHONE_5_IN_8_1_test_bundle
//components/Snackbar:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Snackbar:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Snackbar:unit_tests_IPHONE_X_IN_11_0
//components/Snackbar:unit_tests_IPHONE_X_IN_11_0_test_bundle
```

<!-- identifier: affected-targets -->

material-automation commented:
## API diff detected the following changes

### Buttons

#### MDCButton

*new* property: `mdc_legacyFontScaling` in `MDCButton`

<!-- identifier: apidiff -->

yarneo committed:
[Ripple] Graduate Ripple to Ready. (#7000)

Following updating the docs in #6996 and adding additional tests #6992 for the Ripple:

In this PR I am officially graduating Ripple to ready.

These are the steps taken to do so:
1. Update docs to not show Beta anymore.
2. Move Ripple from the Beta podspec to the main podspec.
3. Removal of MDCCard+Private, MDCCard+Ripple, MDCCardCollectionCell+Private, MDCCardCollectionCell+Ripple, as they were used as connectors between Ripple being a beta component and Card being a ready component.
4. Moving the relevant code in the above files into the Card implementation.
5. Modifying the property `enableBetaBehavior` for Cards to `enableRippleBehavior` as it is no longer a beta behavior.

Tested to see all the examples work well for Cards and CardCollectionCells.

Closes #6941

Commit:
[BottomNavigation] Split snapshot tests. (#6997)

Separating the title layout-related snapshot tests and blur effect tests
to trim down the main tests file.

Part of #6844

jverkoey commented on components/Buttons/examples/ButtonsDynamicTypeViewController.swift:
```diff
@@ -47,29 +47,43 @@ class ButtonsDynamicTypeViewController: UIViewController {
-    flatButtonDynamic.mdc_adjustsFontForContentSizeCategory = true
-    view.addSubview(flatButtonDynamic)
+    containerScheme.typographyScheme = MDCTypographyScheme.init(defaults: .material201902)
+    let flatButtonDynamicM2 = MDCButton()
```
Please consider an alternative suffix :)

ianegordon replied:
Done.

jverkoey commented on components/Buttons/src/MDCButton.h:
```diff
@@ -118,6 +118,13 @@
     BOOL mdc_adjustsFontForContentSizeCategory UI_APPEARANCE_SELECTOR;
 
+/**
+ Enable legacy font scaling curves for Dynamic Type
```
```suggestion
 Enable legacy font scaling curves for Dynamic Type.
```

jverkoey commented on components/Buttons/src/MDCButton.h:
```diff
@@ -118,6 +118,13 @@
 
+/**
+ Enable legacy font scaling curves for Dynamic Type
+
```
Please explain what "legacy font scaling curves" means here.

ianegordon replied:
Done.

jverkoey committed:
Update components/Buttons/src/MDCButton.h

Co-Authored-By: ianegordon <iangordon@google.com>

Commit:
Merge branch 'feature-buttons-dt2' of github.com:material-components/material-components-ios into feature-buttons-dt2

Commit:
Feedback

material-automation commented on components/Buttons/src/MDCButton.m:
```diff
@@ -3,7 +3,7 @@
 // you may not use this file except in compliance with the License.
 // You may obtain a copy of the License at
-//
+  //
```
clang-format suggested the following change:


```
-  //
+//
```

<!-- identifier: clang-format -->
<!-- File: components/Buttons/src/MDCButton.m -->

Commit:
clang-format

ianegordon commented:
CLA warning is not accurate as @jverkoey is a Googler, even if his @google.com email address wasn't attached to his recommendation / commit.

romoore commented:
CLA warning is accurate since the commits came from multiple sources. To ensure the original author (and not only the submitter) wished to merge the PR, the bot sets `cla: No` as a reminder for PR authors to get consent from all authors and document the same.
